### PR TITLE
Launch - Fix up/ add more doc strings

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

- Adds doc strings to launch stuff
- changes git regex
- adds a section to covercircle specific to launch
- removes unused create_run_queue api method

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points. If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/sdk/launch/launch.py [hunk 1]
```diff
@@ -103,21 +103,12 @@ def run(
...
              is unsuccessful.
-    :param uri: URI of project to run. A local filesystem path
-                or a Git repository URI pointing to a project directory containing an MLproject file.
-    :param entry_point: Entry point to run within the project. If no entry point with the specified
-                        name is found, runs the project file ``entry_point`` as a script,
-                        using "python" to run ``.py`` files and the default shell (specified by
-                        environment variable ``$SHELL``) to run ``.sh`` files.
+    :param uri: URI of experiment to run. A wandb run uri or a Git repository URI.
+    :param api: An instance of a wandb Api from wandb.apis.internal.
+    :param entry_point: Entry point to run within the project. Defaults to using the command used to run the original run for wandb URIs, or main.py for git repository URIs.
```
Comment (stephchen) [inline wandb/sdk/launch/launch.py:111]: maybe specify like "defaults to using entry point in the original run"

File: wandb/sdk/launch/launch.py [hunk 2]
```diff
@@ -103,21 +103,12 @@ def run(
...
     :param version: For Git-based projects, either a commit hash or a branch name.
     :param parameters: Parameters (dictionary) for the entry point command.
```
Comment (stephchen) [inline wandb/sdk/launch/launch.py:113]: do we also default to the same parameters as pulled from the original run command if nothing is provided? i actually don't remember

File: wandb/sdk/launch/utils.py [hunk 3]
```diff
@@ -51,8 +51,8 @@ def _is_wandb_local_uri(uri: str) -> bool:
     return _WANDB_LOCAL_DEV_URI_REGEX.match(uri) is not None
 
 
-def _is_github_uri(uri: str) -> bool:
-    return _GIT_URI_REGEX.match(uri) is not None
+def _is_git_uri(uri: str) -> bool:
+    return bool(_GIT_URI_REGEX.match(uri))
```
Comment (stephchen) [inline wandb/sdk/launch/utils.py:55]: what was this change for? just wondering since the wandb regex check is still `is not None`

**Review by stephchen — APPROVED** [review]
ty for doing all this, just a couple little notes

File: wandb/sdk/launch/launch.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/launch.py:113]: yeah, I'll add that

File: wandb/sdk/launch/launch.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/launch.py:111]: I'll make that change

File: wandb/sdk/launch/utils.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/utils.py:55]: This is cleaner, forgot to change the other ones, so I'll change those now

**Pull request merged**